lib/docker-toolbelt.coffee

```diff
@@ -102,6 +102,45 @@ Docker::imageRootDir = (image) ->
 					else
 						throw new Error("Unsupported driver: #{dockerInfo.Driver}/")
 	)
+
+EEXIST = code: 'EEXIST'
+ignore = ->
+MIN_PAGE_SIZE = 4096
+
+aufsMountWithDisposer = (target, layerDiffPaths) ->
+	# We try to create the target directory.
+	# If it exists, it's *probably* from a previous run of this same function,
+	# and the mount will fail if the directory is not empty or something's already mounted there.
+	fs.mkdirAsync(target)
+	.catch code: EEXIST, ignore
+	.then ->
+		appendFromIndex = null
+		options = 'noxino,ro,br='
+		remainingBytes = MIN_PAGE_SIZE - options.length
+		for path, i in layerDiffPaths
```

Comment:
Good idea.  Done (had to tweak a few things for the conversion from `.partition` to using `.findIndex` to work)